### ISSUE TITLE
What's the latest supported Postgres version for GitLab 12.9.2?

### ISSUE THREAD

**Issue by jug** [sameersbn/docker-gitlab#2121]
Thanks for your continuous contributions on this nice image. So far we used this successfully for several years now.

Currently we run GitLab with version sameersbn/gitlab:12.6.4 (soon want to update to the latest 12.9.2).
But now our DB Admins want us to migrate to a newer postgres version with 12.2 being the latest (as writing this post). Currently we still use postgres 9.6.

What is the latest postgres version the gitlab version sameersbn/gitlab:12.9.2 supports?

**Comment by tomislater** [comment]
According to https://about.gitlab.com/releases/2020/03/22/gitlab-12-9-released/:

> To take advantage of improved performance and functionality in PostgreSQL 11, such as partitioning, we plan to require PostgreSQL 11 in GitLab 13.0. PostgreSQL 11 was added in Omnibus GitLab 12.8. It will become the default PostgreSQL version in GitLab 12.10. While PostgreSQL 9.6 and 10.x are still supported and bundled in Omnibus GitLab, they will be removed in GitLab 13.0.

So, you should be able to use `12.9.2` GitLab on `9.6` PostgreSQL.

**Comment by tomislater** [comment]
Btw, I am using `12.8` GitLab and `11.6` PostgreSQL and everything works fine.

**Comment by jug** [comment]
@tomislater Yes, thanks for the link and info with being able to use 11.6 postgres with 12.8 GitLab.
That goes in the right direction ;-)

My question aims also to the point if the latest postgres (e.g. PostgresSQL 12.2) works together with  GitLab 12.6 and GitLab >12.9?

**Comment by tomislater** [comment]
I do not know :/ Officially GitLab currently does not support PostgreSQL 12 as far as I know. But, it does not imply that it does not work.

**Comment by paddy-hack** [comment]
> Btw, I am using 12.8 GitLab and 11.6 PostgreSQL and everything works fine.

That's only true if you don't include your database in the `gitlab:backup:create` task.  See #2092.

**Comment by sachilles** [comment]
I'm runnig gitlab 12.9.3 and PostgreSQL 12.2. However, I'm running gitlab for my own und thus, I can't state that everything is working.

If you want to use `gitlab:backup:create` you need to replace `postgresql-client-10` and `postgresql-contrib-10` by the packages `postgresql-client-12` and `postgresql-contrib-12` in the `Dockerfile`.

**Comment by paddy-hack** [comment]
AFAIK, GitLab.com does not officially support PostgreSQL 12, yet.

As for replacing the `postgresql` packages in the image, you're right.  I've purged `postgresql-10` and `postgresql-client-10` in favour of their version 11 counterparts and backups work fine for me.
BTW, the `postgresql-contrib-*` packages no longer exist as of version 10, other than as pure virtual packages.
